sklearn/model_selection/_search.py

```diff
@@ -598,7 +598,15 @@ def fit(self, X, y=None, groups=None, **fit_params):
             train/test set.
 
         **fit_params : dict of string -> object
-            Parameters passed to the ``fit`` method of the estimator
+            Parameters passed to the ``fit`` method of the estimator.
+
+            Depending on the value, a fit parameter may be split into multiple
+            CV groups along with `X` and `y`. For example, the `sample_weight`
+            parameter to :meth:`sklearn.linear_model.SGDClassifier.fit`
+            is split becuase it's an array-like with the same number of
+            samples as `X` and `y`. On the other hand, the `coef_init`
+            parameter is not split along with `X` and `y`, because it's
+            shape is ``[n_features, n_classes]``.
```

Comment:
Which means that it **will** be split if `n_features == n_samples`, which is one reason we try to avoid such parameters to `fit` and would not recommend their use in `fit_params`.

If you can restructure this to implore the user to only use fit_params for things like sample_weight (and, perhaps, scalars), I would very much appreciate this addition